apps/meteor/app/livechat/server/lib/custom-fields.ts

```diff
@@ -19,7 +19,33 @@ export const validateRequiredCustomFields = (customFields: string[], livechatCus
 	}
 };
 
-export async function setCustomFields({ token, key, value, overwrite }: { key: string; value: string; overwrite: boolean; token: string }) {
+export async function updateContactsCustomFields(contact: ILivechatContact, key: string, value: string, overwrite: boolean): Promise<void> {
+	if (overwrite) {
+		contact.customFields ??= {};
+		contact.customFields[key] = value;
+	} else {
+		contact.conflictingFields ??= [];
+		contact.conflictingFields.push({ field: `customFields.${key}`, value });
```

Comment:
To my understanding: if the function receives override:false it will assign all custom fields as conflicting fields for the visitor, even if the visitor has no customfields at all?

Reply:
If that's true, and there's no more decision being done, think you can construct the query in a way you insert all the things without having to fetch the contact and then update 1b1, like:

```
if (overwrite) 
Object.keys(obj).map((k) => update.$set.customFields[k] = obj[k])
else
update.$addToSet = { conflictingFields: { field, value } }

and then LivechatContacts.updateByVisitorId(visitorId, updates)
```

Do you think that would work?

Reply:
(Ideally the custom fields shouldn't be processed 1b1, but at least this way its still N and not N * X lol)